config/seata-config-core/src/main/java/org/apache/seata/config/CachedConfigurationChangeListener.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.seata.config;
+
+public interface CachedConfigurationChangeListener extends ConfigurationChangeListener {
+
+    ConfigurationCache CONFIGURATION_CACHE = ConfigurationCache.getInstance();
+
+    @Override
+    default void afterEvent(ConfigurationChangeEvent event) {
+        ConfigurationChangeListener listener = (ConfigurationChangeListener)CONFIGURATION_CACHE;
+        listener.onProcessEvent(event);
```

Comment:
```suggestion
   ConfigurationChangeListener afterListener = ConfigurationCache.getInstance();

    @Override
    default void afterEvent(ConfigurationChangeEvent event) {
        afterListener.onProcessEvent(event);
```
这样写是不是更加合适?
Is it more appropriate to write this way?

Reply:
CachedConfigurationChangeListener is an instance that holds state.

Reply:
> CachedConfigurationChangeListener is an instance that holds state.

Since we must notify ConfigurationCache, and ConfigurationCache also implements ConfigurationChangeListener, why do we need to use it to make it strong, instead of defining it in a member variable?
既然必定要通知到ConfigurationCache,而且ConfigurationCache 也是实现了ConfigurationChangeListener,为什么要在使用使强壮,而不再成员变量中定义好?

Reply:
And then I think since it's an afterlistener, shouldn't it just call onChangeEvent instead of onProcessEvent?
然后我认为既然是afterlistener,是否应该直接调用onChangeEvent,而不是onProcessEvent?

Reply:
Design should not be function-oriented but object-oriented; consider design from the perspective of interactive behavior, not at the level of assembling functions. If ConfigurationCache's before and after are defined to be triggered identically, a logical call stack would be: NacosConfiguration.addConfigListener -> event -> listener.beforeEvent -> listener.onChangeEvent -> listener.afterEvent (ConfigurationCache.beforeEvent -> ConfigurationCache.onChangeEvent -> ConfigurationCache.afterEvent)

Reply:
> > CachedConfigurationChangeListener is an instance that holds state.
> 
> Since we must notify ConfigurationCache, and ConfigurationCache also implements ConfigurationChangeListener, why do we need to use it to make it strong, instead of defining it in a member variable? 既然必定要通知到ConfigurationCache,而且ConfigurationCache 也是实现了ConfigurationChangeListener,为什么要在使用使强壮,而不再成员变量中定义好?

Sorry, I'm unable to understand the logic expressed here.

Reply:
> > > CachedConfigurationChangeListener is an instance that holds state.
> > 
> > 
> > Since we must notify ConfigurationCache, and ConfigurationCache also implements ConfigurationChangeListener, why do we need to use it to make it strong, instead of defining it in a member variable? 既然必定要通知到ConfigurationCache,而且ConfigurationCache 也是实现了ConfigurationChangeListener,为什么要在使用使强壮,而不再成员变量中定义好?
> 
> Sorry, I'm unable to understand the logic expressed here.

Why not define it as a ConfigurationChangeListener afterListener = ConfigurationCache.getInstance(); when the member variable is defined, instead of doing the type conversion in afterEvent?
为什么不在成员变量定义的时候就定义为一个ConfigurationChangeListener afterListener = ConfigurationCache.getInstance(); 而是在afterEvent中再进行类型转换?

Reply:
> Design should not be function-oriented but object-oriented; consider design from the perspective of interactive behavior, not at the level of assembling functions. If ConfigurationCache's before and after are defined to be triggered identically, a logical call stack would be: NacosConfiguration.addConfigListener -> event -> listener.beforeEvent -> listener.onChangeEvent -> listener.afterEvent (ConfigurationCache.beforeEvent -> ConfigurationCache.onChangeEvent -> ConfigurationCache.afterEvent)

我是认为ConfigurationCache不需要再有beforeEvent的相关逻辑了,如果有的话我认为是否未来可能会出现套娃现象?
I was thinking that ConfigurationCache doesn't need to have beforeEvent related logic anymore, and if it does do I think there might be nesting in the future?

Reply:
> > Design should not be function-oriented but object-oriented; consider design from the perspective of interactive behavior, not at the level of assembling functions. If ConfigurationCache's before and after are defined to be triggered identically, a logical call stack would be: NacosConfiguration.addConfigListener -> event -> listener.beforeEvent -> listener.onChangeEvent -> listener.afterEvent (ConfigurationCache.beforeEvent -> ConfigurationCache.onChangeEvent -> ConfigurationCache.afterEvent)
> 
> 我是认为ConfigurationCache不需要再有beforeEvent的相关逻辑了,如果有的话我认为是否未来可能会出现套娃现象? I was thinking that ConfigurationCache doesn't need to have beforeEvent related logic anymore, and if it does do I think there might be nesting in the future?

This is about extendable design; it's not about whether a feature exists or not. If it doesn't exist, you can maintain an empty implementation, and if there is predefined logic, you can add it as needed. I suggest that you should not judge the design solely based on current functionalities. For example, with cache, aspects like secret computing and storage are points that can be extended both before and after.

Reply:
> > > > CachedConfigurationChangeListener is an instance that holds state.
> > > 
> > > 
> > > Since we must notify ConfigurationCache, and ConfigurationCache also implements ConfigurationChangeListener, why do we need to use it to make it strong, instead of defining it in a member variable? 既然必定要通知到ConfigurationCache,而且ConfigurationCache 也是实现了ConfigurationChangeListener,为什么要在使用使强壮,而不再成员变量中定义好?
> > 
> > 
> > Sorry, I'm unable to understand the logic expressed here.
> 
> Why not define it as a ConfigurationChangeListener afterListener = ConfigurationCache.getInstance(); when the member variable is defined, instead of doing the type conversion in afterEvent? 为什么不在成员变量定义的时候就定义为一个ConfigurationChangeListener afterListener = ConfigurationCache.getInstance(); 而是在afterEvent中再进行类型转换?

I understand what you mean. However, this might be related to my personal habits. It can be replaced with one line syntactically, but in reality, it involves two steps. 
But from a design perspective, I would like to point out a few things. Member variables are intended for internal use within a class to maintain consistency in the architecture. Once converted into an interface here, it means that other methods can only use methods defined by the interface.

Reply:
ok